istioctl/cmd/analyze.go

```diff
@@ -271,3 +273,36 @@ func errorIfMessagesExceedThreshold(messages []diag.Message) error {
 
 	return nil
 }
+
+type thresholdLevelParser struct {
+	diag.Level
+}
+
+// String satisfies interface pflag.Value
+func (m *thresholdLevelParser) String() string {
+	return m.Level.String()
+}
+
+// Type satisfies interface pflag.Value
+func (m *thresholdLevelParser) Type() string {
+	return "Level"
+}
+
+// Set satisfies interface pflag.Value
+func (m *thresholdLevelParser) Set(s string) error {
+	l, err := LevelFromString(s)
+	if err != nil {
+		return err
+	}
+	m.Level = l
+	return nil
+}
+
+func LevelFromString(s string) (diag.Level, error) {
+	val, ok := diag.GetUppercaseStringToLevelMap()[strings.ToUpper(s)]
+	if !ok {
+		return diag.Level{}, fmt.Errorf("%q not a valid option, please choose from: %v", s, diag.GetAllLevelStrings())
+	}
+
+	return val, nil
+}
```

Comment:
If we ever need to do string->level translation anywhere else it will make sense for this to live in level.go instead. I have a very slight preference for having it there in the first place, but I'm OK leaving it here for now as well.

Reply:
I agree, but in that case I think we need to do a bit more work instead of just assuming that the strings will come in uppercase all of the time so it'll need another PR anyways.